app/views/SettingsView/index.js

```diff
@@ -87,15 +89,33 @@ class SettingsView extends React.Component {
 		return roles?.includes('livechat-agent');
 	}
 
+	logout = () => {
+		const { logout, user } = this.props;
+		if (!user.loginEmailPassword) {
```

Comment:
Following my comments, this would be a user query instead of Redux.
It's going to make the code on this call slightly complex, but will simplify the rest of the code a lot.